src/common/str_utils.cpp

```diff
@@ -261,6 +261,22 @@ void StringBuilder::append_string(const char *str) {
     *current_pos_ = '\0';
 }
 
+void StringBuilder::append_string_view(string_view_utf8 str) {
+
+    while (true) {
+        if (is_problem()) {
+            return;
+        }
+
+        char b = str.getUtf8Char();
```

Comment:
getUtf8Char does decoding the string and returns a full unicode. You cannot use it here. You have to use getbyte

Reply:
getbyte was private, so I added a public getbyte function